peernotifier/peernotifier.go

```diff
@@ -0,0 +1,92 @@
+package peernotifier
+
+import (
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/subscribe"
+)
+
+// PeerNotifier is a subsystem which observes peer offline and online events.
+// It takes subscriptions for its events, and whenever it observes a new event
+// it notifies its subscribers over the proper channel.
+type PeerNotifier struct {
+	started sync.Once
+	stopped sync.Once
+
+	ntfnServer *subscribe.Server
+}
+
+// PeerOnlineEvent represents a new event where a peer comes online.
+type PeerOnlineEvent struct {
+	// PubKey is the ecdsa.PubKey of the peer that came online.
+	PubKey *btcec.PublicKey
```

Comment:
we may want to consider making these just `[33]byte` arrays, since for the most i think we only need to use this as an identifier. not a block change though, it's something we can consider in the future

Reply:
Agreed that it's more likely to be used as an identifier, so I've changed to `[33]byte`. Extra info can always be added if it's needed, but I don't see much of a case for it as is.